### DISCUSSION TITLE
Attach create multiple records in relationship table

### DISCUSSION THREAD

**Discussion by LeCoupa** [adonisjs/core#1974] [Help] [answered]
In Adonis 4, in a Many to Many Relationship, the attach method was not duplicating records in the relationship table.

It seems like this behavior has changed in Adonis 5. Executing this code below will create two records in `model1_model2` (even when there is already an existing relation).

```javascript
model1 = Model1.find(...)
model2 = Model2.find(...)

model1.attach([model2.id]) <-- insert a record in model1_model2
model1.attach([model2.id])<-- insert anoter record in model1_model2 (which was not the case in Adonis4)
```

Is there a way to go back to the previous behavior? I would like to avoid having duplicated records in my database or to have to check before every attach method if there is already a relationship between the two models.

**Comment by thetutlage** [comment, marked as answer]
You must use the `sync` method and not `attach`. I am not sure if `attach` worked the way you are explaining in v4. The `sync` method is the one that avoids the duplicates

**Reply by LeCoupa** [reply]
Yes, the issue with the `sync` method in my case is that it will remove all other relationships. So if I am not mistaken, now when I want to add a new relationship without duplication, I must check if the relationship exists before attaching it.

**Reply by thetutlage** [reply]
Nope, it will not. Recommend following the docs always https://preview.adonisjs.com/guides/model-relations/many-to-many#sync

**Reply by popovicieduard** [reply]
Also, sync has an additional detach boolean, I think this is what @thetutlage was reffering to:
```
sync(ids: (string | number)[] | {
  [key: string]: ModelObject;
 }, detach?: boolean, trx?: TransactionClientContract): Promise<void>;
```

**Reply by LeCoupa** [reply]
@popovicieduard thank you! This is exactly what I was looking for as detach is `true` by default. 🙏
